pandas/plotting/_matplotlib/hist.py

```diff
@@ -81,8 +111,22 @@ def _make_plot(self):
                 kwds["weights"] = weights[:, i]
 
             artists = self._plot(ax, y, column_num=i, stacking_id=stacking_id, **kwds)
+
+            # when by is applied, show title for subplots to know which group it is
+            if self.by is not None:
+                ax.set_title(pprint_thing(label))
+
             self._add_legend_handle(artists[0], label, index=i)
 
+    def _reformat_y(self, y: Union[Series, np.array]) -> Union[Series, np.array]:
+        """Internal function to reformat y given `by` is applied or not."""
```

Comment:
Can you clarify this? I think known from the function name that it is internal to reformat y, but curious what the existence of by should change?

Reply:
sure, main difference is the dimension of y will be different, if by is None, y is 1-d array, while if by is not None, which means group by will happen, and y here is multi-dimension array.

i will change to a more detailed description for this.